executor/src/main/java/com/redhat/service/bridge/executor/K8SDeploymentManagerMock.java

```diff
@@ -0,0 +1,9 @@
+package com.redhat.service.bridge.executor;
+
+import com.redhat.service.bridge.infra.dto.ProcessorDTO;
+
+public interface K8SDeploymentManagerMock {
```

Comment:
@danielezonca I don't think so. I'll rename `K8SDeploymentManagerMock` to `ExecutorDeploymentManager` because it might help to get the right idea. 

The application configuration is Injected by the bean that implements `ExecutorConfigProvider` (I will rename it soon). Until we use the all-in-one application we provide a dummy implementation of `ExecutorConfigProvider`: `ExecutorConfigProviderMock`. The `ExecutorConfigProviderMock` implements `K8SDeploymentManagerMock` as well, so that we can include the "deployment" logic behind the scenes/without touching any core part of the executor.
On the other side, the `k8s` module will inject a  `K8SDeploymentManagerMock` bean, so to control the deployment/undeployment of the resources.

Reply:
In other words, `K8SDeploymentManagerMock` is the bean that must stay in the module that has to be deployed/undeployed

Reply:
Sorry @danielezonca , now I got what you meant: you asked to move this interface to the k8s module and we keep the bean that implements it in the executor. 
At the moment the dependency is `k8s` -> `executor`. 
It would be ok to invert it, but the `shard` atm imports `k8s` as dependency and uses it in some tests: we would have to provide a mock for that bean and do some other stuff to make it work. 

Since the entire k8s module will be thrown away soonish I think it's fine as it is, but if the switch to k8s will take longer we can work on https://issues.redhat.com/browse/MGDOBR-48